CONTRIBUTING.md

```diff
@@ -82,7 +82,7 @@ a detailed commit message body is preferred. Make sure to keep the `Signed-off-b
 
 - Python 3.11
 - Go 1.21
-- JDK 17
+- JDK 19
```

Comment:
I haven't checked with JDK 17. It's possible the builds are green but the Gradle docs recommend Java version 19 for running Gradle 7.6 : https://docs.gradle.org/current/userguide/compatibility.html#java_runtime

I'm guessing Gradle is pinned to 7.6 on purpose right?

Reply:
Note: I'm no longer an official member of the Macaron team. I just accidentally came across this ticket and I'm just trying to give my two cents here.

Thanks for checking the Gradle docs. However, I think you may be misinterpreting the Java Compatibility table.

The section you linked to says:

>  A JVM version between 8 and 23 is required to execute Gradle. JVM 24 and later versions are not yet supported.

> Executing the Gradle daemon with JVM 16 or earlier has been deprecated and will become an error in Gradle 9.0. The Gradle wrapper, Gradle client, Tooling API client, and TestKit client will remain compatible with JVM 8.

From these details, the Java Compatibility table can be interpreted as follows

* Java 19 is compatible with Gradle 7.6 and onwards. It does not mean Gradle 7.6 should only be used or is recommended to be used with Java 19.
* Key components of Gradle (the wrapper, the client, etc.) will continue to be compatible with Java 8. The Gradle daemon is one component of Gradle that will not be compatible with Java earlier than 17 coming in Gradle version 9.0. However, it is an optional component of Gradle, meant for speeding up incremental builds, and one can still run Gradle in no-daemon mode if they need to.

Other than that, requiring Java 17 in particular is a sensible choice as it is an [LTS Java version](https://www.oracle.com/au/java/technologies/java-se-support-roadmap.html). One advantage of using it over a non-LTS one like Java 19 is that it generally has better support. For example, JDK 8, 11, 17, 21, should be mostly available in the official package repositories of most Linux distributions, while the same thing may be untrue for non-LTS versions.

Really appreciate you having a look at these things and trying to improve the development experience for contributors. This is something I know the team cares about deeply. However, for future changes, I would recommend to first have initial discussions with the team about your change, in an issue for instance, before going ahead with PRs. Changing the required Java version, for example, is not a straightforward change. If you look closely, to fully change the required Java version in the project, the following places also need change:
* [The container image](https://github.com/oracle/macaron/blob/e19c4b2a476c9fc127fb69603f8e76e3132b9764/docker/Dockerfile.base#L76)
* [Our GitHub Actions CI workflow](https://github.com/oracle/macaron/blob/staging/.github/workflows/_build.yaml#L69)

Having a good idea of what should and should not be changed before a PR will help the team help you in creating PRs and save time for all of us so that we can focus on more interesting tasks. Thanks for reading this long message :slightly_smiling_face:.

Reply:
My apologies for the noise. I'm merely using this PR as a way to track the changes I've had to make to get the project running locally. Ofcourse, early reviews are helpful but since the PR is marked as "Draft", I wasn't expecting any eyeballs yet.

Thanks for looking into this anyway. I hadn't checked the Java LTS releases so sticking to Java 17 definitely makes sense. More than happy to revert this.

Reply:
reverted in https://github.com/oracle/macaron/pull/841/commits/85446ff883310175496b40faa5d70738f344b8e3